off-chain/src/Network/OrderedMap.purs

```diff
@@ -43,4 +59,7 @@ splitEither
 splitEither = undefined
 
 empty :: forall a b. OrderedMap a b
-empty = undefined
+empty = OrderedMap { map: Map.empty, array: [] }
+
+orderedElems :: forall k v. OrderedMap k v -> Array (k /\ v)
+orderedElems (OrderedMap oMap) = Array.reverse oMap.array
```

Comment:
why is this reversing the order of the array?

Reply:
I've used `cons` in `push` for array part. Changed it to `snoc`.